pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -213,12 +214,17 @@ func setupGC(t *testing.T, config *restclient.Config) garbageCollector {
 	if err != nil {
 		t.Fatal(err)
 	}
+	tweakableRM := meta.NewDefaultRESTMapper(nil)
+	rm := &testRESTMapper{meta.MultiRESTMapper{tweakableRM, testrestmapper.TestOnlyStaticRESTMapper(legacyscheme.Scheme)}}
 
 	client := fake.NewSimpleClientset()
 	sharedInformers := informers.NewSharedInformerFactory(client, 0)
+	metadataInformers := metadatainformer.NewSharedInformerFactory(metadataClient, 0)
 	alwaysStarted := make(chan struct{})
 	close(alwaysStarted)
-	gc, err := NewGarbageCollector(ctx, client, metadataClient, &testRESTMapper{testrestmapper.TestOnlyStaticRESTMapper(legacyscheme.Scheme)}, ignoredResources, sharedInformers, alwaysStarted)
+
+	gc, err := NewGarbageCollector(ctx, client, metadataClient, rm, map[schema.GroupResource]struct{}{},
+		informerfactory.NewInformerFactory(sharedInformers, metadataInformers), alwaysStarted)
```

Comment:
I don't understand why the informers changed.  If we are unable to use the previous logic, then we broke something.  This is a blocker for merge.

Reply:
reverted to original